Mathlib/SetTheory/Lists.lean

```diff
@@ -176,9 +176,10 @@ theorem mem_of_subset' {a} : ∀ {l₁ l₂ : Lists' α true} (_ : l₁ ⊆ l₂
 theorem subset_def {l₁ l₂ : Lists' α true} : l₁ ⊆ l₂ ↔ ∀ a ∈ l₁.toList, a ∈ l₂ :=
   ⟨fun H _ => mem_of_subset' H, fun H => by
     rw [← of_toList l₁]
-    revert H; induction' toList l₁ with h t t_ih <;> intro H
-    · exact Subset.nil
-    · simp only [ofList, List.find?, List.mem_cons, forall_eq_or_imp] at *
+    revert H; induction toList l₁ with intro H
```

Comment:
```suggestion
    induction toList l₁ generalizing H with
```

Reply:
This doesn't work. The goal state at the start of the `cons` branch is
```
α : Type u_1
l₁ l₂ : Lists' α true
h : Lists α
t : List (Lists α)
t_ih : (∀ a ∈ l₁.toList, a ∈ l₂) → ofList t ⊆ l₂
H : ∀ a ∈ l₁.toList, a ∈ l₂
⊢ ofList (h :: t) ⊆ l₂
```
There's no lemma generated saying `l₁.toList = h :: t`. If instead I try `induction z : toList l₁ generalizing H with`, `t_ih`'s type does not match:
```
α : Type u_1
l₁ l₂ : Lists' α true
h : Lists α
t : List (Lists α)
t_ih : (∀ a ∈ l₁.toList, a ∈ l₂) → l₁.toList = t → ofList t ⊆ l₂
H : ∀ a ∈ l₁.toList, a ∈ l₂
z : l₁.toList = h :: t
⊢ ofList (h :: t) ⊆ l₂
```
But the other two have been fixed.